paasta_tools/kubernetes_tools.py

```diff
@@ -1516,8 +1516,63 @@ def get_pod_volumes(
         if crypto_volume:
             pod_volumes.append(crypto_volume)
 
+        database_credentials_secrets_volumes = (
+            self.get_database_credentials_secrets_volumes()
+        )
+        if database_credentials_secrets_volumes:
+            # note an array is returned so we extend vs append
+            pod_volumes.extend(database_credentials_secrets_volumes)
```

Comment:
not saying this is correct, but @flopex and I figured it's best to keep the secrets in separate Kubernetes secret configs and map each individually. this also allows each secret to have it's own nested path (ie `/dstore/<mysql or cassandra>/<credential name>/<secret path>)

might be wrong, starting to forget the rationale but open to changing it if adding too many volumes is incorrect

if this just means we turn it into a for loop like in the secret_volumes, that is fair too. extend was just cleaner

Reply:
afaik, we should still be able to give these their own paths using https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#project-secret-keys-to-specific-file-paths

if possible, it'd be nice to continue the 1 volumes -> N secrets that we do for most secrets (secret_volumes is an outlier here, but that's because every secret there can have a different mount path)

my understanding is that we should be able set the `mountPath` config to `/datastore` and then every secret that's mounted in `items[]` uses the `path` config to finish out the `/datastore/.../...` path

Reply:
i.e., my concern isn't extend() vs append() - but having N volumes vs just 1 overall

Reply:
happy to refactor! let me know which is preferred

Reply:
++ refactored as requested